specs/sequencer-relayer.md

```diff
@@ -0,0 +1,152 @@
+# Sequencer-Relayer
+
+## Overview
+
+The sequencer-relayer, a.k.a relayer, is an application whose primary
+responsibility is to store sequenced rollup data on a remote data availability
+layer (DA layer).
+
+The immediate source of the sequenced data is the sequencer blockchain network,
+where rollups' data is sequenced and included in sequencer blocks, while the DA
+layer is the Celestia blockchain.
+
+## Application Logic
+
+The application logic of the relayer is comprised of three main tasks: getting
+the data from a sequencer node, transforming the data in preparation for
+storing, and putting the transformed data onto the Celestia network.
+
+### Getting the Data from a Sequencer Node
+
+The relayer runs an endlessly repeating task, the "reader task", which does the
+following:
+
+1. polls the sequencer node for the latest (highest) sequencer block height at
+  a fixed frequency (once per second currently)
+1. upon learning of a new height, fetches the block at that height from the
+  sequencer node
+1. upon receiving the sequencer block, forwards it to the transformation task
+
+Note that currently the relayer only communicates with a single sequencer node,
+and that this node is comprised of two separate processes, each providing their
+own http servers. Consequently, the relayer queries two different endpoints
+using two different protocols; CometBFT's ABCI `Info` for the block height via
+JSONRPC, and the sequencer app's `GetSequencerBlock` via gRPC. This is likely to
+improve in the near future, as work is under way to extend the sequencer app's
+API to the extent that no direct communication with the CometBFT node will be
+required.
+
+Should the sequencer fail to fetch the latest sequencer block height, no action
+is taken other than logging and recording the fact in metrics - polling
+continues at the same frequency under tha assumption that the error is a
```

Comment:
```suggestion
continues at the same frequency under the assumption that the error is a
```

Reply:
Well spotted! Fixed in [045e82c](https://github.com/astriaorg/astria/pull/1126/commits/045e82c6be5afcd9e2fd53055e3d977e4037ecbf).